src/method.jl

```diff
@@ -59,22 +60,54 @@ end
 
 """
     percival(nlp)
+
+A matrix-free augmented Lagrangian for nonlinear optimization.
+
+For advanced usage, first define a `PercivalSolver` to preallocate the memory used in the algorithm, and then call `solve!`:
+
     solver = PercivalSolver(nlp)
     solve!(solver, nlp)
 
-Implementation of an augmented Lagrangian method. The following keyword parameters can be passed:
-- x: the initial guess (default: nlp.meta.x0).
-- μ: Starting value of the penalty parameter (default: 10.0)
-- atol: Absolute tolerance used in dual feasibility measure (default: 1e-8)
-- rtol: Relative tolerance used in dual feasibility measure (default: 1e-8)
-- ctol: (Absolute) tolerance used in primal feasibility measure (default: 1e-8)
-- max_iter: Maximum number of iterations (default: 1000)
-- max_time: Maximum elapsed time in seconds (default: 30.0)
-- max_eval: Maximum number of objective function evaluations (default: 100000)
-- subsolver_logger: Logger passed to `tron` (default: NullLogger)
-- inity: Initial values of the Lagrangian multipliers
-- subsolver_kwargs: subsolver keyword arguments as a dictionary
-- `verbose::Integer = 0`: if > 0, display iteration details every `verbose` iteration.
+# Arguments
+- `nlp::AbstractNLPModel{T, V}` is the model to solve, see `NLPModels.jl`.
+
+# Keyword arguments 
+- `x::V = nlp.meta.x0`: the initial guess.
+- `atol::T = T(1e-8)`: absolute tolerance.
+- `rtol::T = T(1e-8)`: relative tolerance: algorithm stops when ‖∇f(xᵏ)‖ ≤ atol + rtol * ‖∇f(x⁰)‖.
```

Comment:
There are constraints, so this stopping test isn't right.

Reply:
Thanks for the comment. It is fixed.